fineract-provider/src/main/java/org/apache/fineract/infrastructure/creditbureau/service/CreditReportWritePlatformServiceImpl.java

```diff
@@ -202,18 +202,15 @@ public CommandProcessingResult deleteCreditReport(Long creditBureauId, JsonComma
             try {
                 this.creditReportRepository.delete(creditReport);
             } catch (final JpaSystemException | DataIntegrityViolationException dve) {
-                throw new PlatformDataIntegrityException("error.msg.cund.unknown.data.integrity.issue",
-                        "Unknown data integrity issue with resource: " + dve.getMostSpecificCause(), dve);
+                throw ErrorHandler.getMappable(dve, "error.msg.cund.unknown.data.integrity.issue",
+                        "Unknown data integrity issue with resource: " + dve.getMostSpecificCause().getMessage());
```

Comment:
The mostSpecificCause() can be null, so here an NPE will happen... i think we should be careful to avoid that!

Reply:
It can not be null, if there is no cause, then the same instance is returned.

Reply:
My bad...